src/idl/src/tree.c

```diff
@@ -3502,10 +3518,64 @@ bool idl_is_topic_key(const void *node, bool keylist, const idl_path_t *path, ui
 
 bool idl_is_extensible(const idl_node_t *node, idl_extensibility_t extensibility)
 {
-  return (idl_is_struct(node) && ((idl_struct_t *)node)->extensibility.value == extensibility) ||
-    (idl_is_union(node) && ((idl_union_t *)node)->extensibility.value == extensibility);
+  if (idl_is_struct(node)) {
+    const idl_struct_t *_struct = (const idl_struct_t *)node;
+    return _struct->extensibility.value == extensibility;
+  }
+  if (idl_is_union(node)) {
+    const idl_union_t *_union = (const idl_union_t *)node;
+    return _union->extensibility.value == extensibility;
+  }
+  return false;
 }
 
+bool idl_has_implicit_default_extensibility(const idl_node_t *node)
+{
+  assert(node);
+  for (; node; node = idl_next(node)) {
+    if (idl_mask(node) == IDL_MODULE) {
+      const idl_module_t *module = (const idl_module_t *)node;
+      return idl_has_implicit_default_extensibility(module->definitions);
+    } else if (idl_mask(node) == IDL_STRUCT) {
+      const idl_struct_t *_struct = (const idl_struct_t *)node;
+      if (!_struct->extensibility.annotation && !_struct->extensibility.implicit)
```

Comment:
I find it surprising that for this function to return `true` (that is, "implicit"), the `implicit` field has to be false. There's got to be a better solution!

Reply:
You're right, that's confusing! I've refactored this in 93447b3